tools/addrundata/rundata.go

```diff
@@ -56,6 +61,32 @@ func (pd *parsedData) fromCode(r io.Reader) error {
 	if err := sc.Err(); err != nil {
 		return err
 	}
+	if !foundInit {
+		return errors.New("missing func init()")
+	}
+	return nil
+}
+
+// fromCode reads parsedData from a textproto.
+func (pd *parsedData) fromProto(r io.Reader) error {
+	bytes, err := io.ReadAll(r)
+	if err != nil {
+		return err
+	}
+	md := new(mpb.Metadata)
+	if err := prototext.Unmarshal(bytes, md); err != nil {
+		return err
+	}
+	if uuid := md.GetUuid(); uuid != "" {
+		pd.testUUID = uuid
+	}
+	if planID := md.GetPlanId(); planID != "" {
+		pd.testPlanID = planID
+	}
+	if desc := md.GetDescription(); desc != "" {
+		pd.testDescription = desc
+	}
```

Comment:
This initializes pd, so we don't need to care about setting fields conditionally.

```suggestion
	pd.testUUID = md.GetUuid()
	pd.testPlanID = md.GetPlanId()
	pd.testDescription = md.GetDescription()
```